site/it/community/contribute/code_style.md

```diff
@@ -0,0 +1,164 @@
+$ clang-format <my_cc_file> --style=google > /tmp/my_cc_file.cc
+$ diff <my_cc_file> /tmp/my_cc_file.cc
+```
+
+## Altri linguaggi
+
+*   [Guida di Stile Google per Java](https://google.github.io/styleguide/javaguide.html)
+*   [Guida di Stile Google per JavaScript](https://google.github.io/styleguide/jsguide.html)
+*   [Guida di Stile Google per Shell](https://google.github.io/styleguide/shell.xml)
+*   [Guida di Stile Google per Objective-C](https://google.github.io/styleguide/objcguide.html)
+
+
+
+
+## Convenzioni ed usi speciali in TensorFlow
+
+### Tensori
+
+*   Le operazioni che hanno a che trattano batch possono ipotizzare che la **prima dimensione** di
+    un Tensore sia la dimensione del batch.
+*   In molti modelli, l'**ultima dimensione** è il numero di _canali_.
+*   Le dsimensioni escluse la prima e l'ultima, di solito, rappresentano le dimensioni dell _spazio_
+    : la lunghezza della sequenza, o la dimensione dell'immagine.
+    *   Quando possibile, usate un operatore sovraccarivcato tra Tensori, invece di una funzione di TensorFlow. 
+    Per esempio, noi preferiamo `**`, `+`, `/`, `*`, `-`, `and/or` a
+    `tf.pow`, `tf.add`, `tf.divide`, `tf.multiply`, `tf.subtract`, e `tf.logical_*` —
+    a meno che sia richiesto uno specifico nome per l'operazione.
+
+
+### Operazioni Python
+
+Un' _operazione Python_ è una funzione che, dati in input tensori e parametri,
+crea una parte del grafo, e ritorna come risultato dei tensori.
+*   Il primo argomento dovrà essere un tensore, seguito dai parametri Python di base.
+    L'ultimo argomento è `name` con valore di default `None`.
+*   Gli argomenti TensorFlow dovrebbero essere un tensore singolo o un elenco "iterabile" di 
+    tensori. Perché, "un Tensore o una lista di Tensori" è troppo generico. Vedere `assert_proper_iterable`.
+*   Le opreazioni che prendono tensori come argomenti ed usano operazioni C++, 
+    dovranno chiamare `convert_to_tensor` per convertire input non-tensori in tensori.
+    Notare che gli argomenti sono acora descritti nella documentazione come oggetti 
+    di uno specifico tipo dtype: `Tensor`.
+*   Ogni operazione Python dovrebbe avere un `name_scope`. 
+    Come si può vedere sotto, per passare il nome dell'operaizone come una stringa.
+*   Le operazioni dovrebbero contenere un commento estensibile Python con dichiarazioni
+    Args e Returns che spieghino sia il tipo sia il significato di ogni valore. Eventuali
+    shape, dtype, o rank dovrebbero essere specificati nella descrizione. Vedere i
+    dettagli della documentazione.
+*   Per una migliore usabilità, includere un esempio di utilizzo dell'operazione con input / output
+    in una sezione Example.
+*   Evitate di fare uso esplicito di `tf.Tensor.eval` o `tf.Session.run`. Per esempio,
+    per scrivere della logica che dipnede dal valore del Tensore, usate il controllo di flusso di TensorFlow.
+    In alternativa, restringete l'operazione a funzionare solo quando è abilitata l'esecuzione eager
```

Comment:
While reading I think "per funzionare" instead of "a funzionare". Does this suggestion make sense?